pvanalytics/quality/gaps.py

```diff
@@ -135,3 +136,106 @@ def interpolation_diff(x, window=3, rtol=1e-5, atol=1e-8):
         atol=atol
     )
     return flags
+
+
+def valid_between(series, days=10, minimum_hours=7.75, freq=None):
+    """Get the start and end of valid data.
+
+    The start and end dates returned by this function can be used to
+    remove large periods of missing data from the begining and end of
+    the series. The valid data begins when there are `days`
+    consecutive days with valid data covering at least `minimum_hours`
+    on each day. Valid data ends on the last day with `days`
+    consecutive days with data covering at least `minimum_hours`
+    preceeding it.
+
+    Any data point with a value other than `NaN` is considered valid
```

Comment:
Since we are only removing `NaN`, "valid" is a bit strong.

I would use "valid" for data that also passes physical limit and consistency checks, for example. I also wouldn't use "complete" or "serially complete" since I would expect the output to not have _any_ missing values. Maybe "sufficient"? 

If `series` is 30 days, and there are lots of missing data on day 1, day 15, and day 30, what is returned?  Day 2, Day 14?  Or something else?

Reply:
I think the important thing to describe is what it means for a value to be missing rather than what it means for it to be 'valid.' You are right, "valid" is not the best word to use here.

> If series is 30 days, and there are lots of missing data on day 1, day 15, and day 30, what is returned? Day 2, Day 14? Or something else?

With the default parameters the returned mask will select day 2-29. If there is missing data in the middle of a series nothing happens to it.